### DISCUSSION TITLE
[MinimalLib] object with multiple molecules

### DISCUSSION THREAD

**Discussion by apahl** [rdkit/rdkit#4293] [Q&A] [answered]
Hi,  
using the RDKIT CFFI bindings in Go I generated a molecule from this two-molecule SMILES: "c1ccccc1CCNC.Cl"
I had expected that the corresponding JSON object would contain two molecules as well, but it contains only one molecule, with the chloride (index: 10) (correctly) not being connected:
```
{"commonchem":{"version":10},"defaults":{"atom":{"chg":0,"impHs":0,"isotope":0,"nRad":0,"stereo":"unspecified","z":6},"bond":{"bo":1,"stereo":"unspecified"}},
"molecules":
    [
        {"atoms":
            [
                {"impHs":1},
                {"impHs":1},
                {"impHs":1},
                {"impHs":1},
                {"impHs":1},
                {},
                {"impHs":2},
                {"impHs":2},
                {"impHs":1,"z":7},
                {"impHs":3},
                {"impHs":1,"z":17}
            ],
            "bonds":
                [
                    {"atoms":[0,1],"bo":2},{"atoms":[1,2]},{"atoms":[2,3],"bo":2},{"atoms":[3,4]},{"atoms":[4,5],"bo":2},{"atoms":[5,6]},{"atoms":[6,7]},{"atoms":[7,8]},{"atoms":[8,9]},{"atoms":[5,0]}
                ],
            "extensions":
                [{"aromaticAtoms":[0,1,2,3,4,5],"aromaticBonds":[0,1,2,3,4,9],"atomRings":[[0,5,4,3,2,1]],"formatVersion":2,"name":"rdkitRepresentation","toolkitVersion":"2021.09.1pre"}
            ]
        }
    ]
}
```
Is the output shown above the expected result? How would I get a JSON object with multiple molecules?

Many thanks in advance.

Kind regards,  
Axel

**Comment by greglandrum** [comment, marked as answer]
@apahl : parsing a SMILES (or Mol block, or Mol2 file, or PDB, or...) always produces a single RDKit molecule. That molecule may be made up of multiple disconnected fragments, but it's still a single molecule.

The way you split a molecule with multiple fragments into multiple molecules in the RDKit is using `GetMolFrags()`, but this isn't currently exposed to the CFFI wrappers. It would clearly be something useful, once we figure out a straightforward way of dealing with returning multiple molecules.

In your particular case: if you really want both molecules, you can always split the SMILES on `.` from Go and then work with the individual sub SMILES. If you just want the larger of the two fragments and don't mind that the molecule is standardized, you can use the `fragment_parent()` function which is in the CFFI wrapper; that will get you a standardized version of the largest fragment.

**Reply by apahl** [reply]
Hi Greg, thanks a lot for the detailed answer.  
I was actually trying to implement the LargestFragment functionality and did not notice that it is already available in the CFFI via the `fragment_parent()` function, this is perfect.

Thanks a lot again for this awesome project!

Kind regards,  
Axel

**Comment by chemphys** [comment]
Hello! Regarding this, is there any way to combine different mol objects into a single mol object? For example, If I have a mol object with the ligand, and a mol object with the protein, that I previously obtained by splitting it with `GetMolFragments`, how do I merge them back together? 

The reason I ask is because I want to remove a given moleculecule from the system, and this seems the easiest way to do it: split, get all molecules that are not the one you wanna remove, and merge everything BUT that molecule. Any advice on that?
